lexos/models/topword_model.py

```diff
@@ -0,0 +1,317 @@
+        group_sums = [np.sum(row, axis=0) for row in group_matrices]
+
+        # Find the comparison map, which is a list of tuples.
+        # There are two elements in each tuple, each one is a index of groups.
+        # Ex: first group has index 0. And two group indexes cannot be equal.
+        comp_map = itertools.product(list(range(len(group_sums))),
+                                     list(range(len(group_sums))))
+        comp_map = [(i_index, j_index)
+                    for (i_index, j_index) in comp_map if i_index != j_index]
+
+        # Compare each paragraph in group_comp to group_base.
+        for comp_index, base_index in comp_map:
+            comp_para = group_matrices[comp_index]
+
+            # Generate analysis result.
+            temp_result_list = [TopwordModel._z_test_word_list(
+                count_list_i=paras,
+                count_list_j=group_sums[base_index],
+                words=self._doc_term_matrix.columns.values)
+                for para_index, paras in enumerate(comp_para)]
+
+            # Attach readable name to each result series.
+            temp_readable_result = [temp_result_list[index].rename(
+                'Document "' + group_file_labels[comp_index][index] +
+                '" compared to Class "' + class_labels[base_index] + '"')
+                for index, _ in enumerate(comp_para)]
+
+            # Put all temp result together.
+            readable_result += temp_readable_result
+
+        return readable_result
+
+    def _analyze_class_to_class(self, division_map: pd.DataFrame) -> \
+            ReadableResult:
+        """Detect if a given word is an anomaly.
+
+        While doing so, this method compares the occurrence of a given word
+        in a class of segments to the occurrence of the same word in other
+        class of segments.
+        :return: a list of series, each series has a readable name and the
+                 result, which contains words with corresponding z-scores.
+        """
+        # Trap possible empty input error.
+        assert not self._doc_term_matrix.empty, SEG_NON_POSITIVE_MESSAGE
+
+        # Initialize all the class labels.
+        class_labels = division_map.index.values
+        # Match labels and word counts into groups.
+        group_matrices = [self._doc_term_matrix.values[row]
+                          for row in division_map.values]
+        # Find the total word count of each group.
+        group_sums = [np.sum(row, axis=0) for row in group_matrices]
+
+        # Find the comparison map, which is a list of tuples.
+        # There are two elements in each tuple, each one is a index of groups.
+        # Ex: first group has index 0. And two group indexes cannot be equal.
+        comp_map = itertools.product(list(range(len(group_sums))),
+                                     list(range(len(group_sums))))
+        comp_map = [(i_index, j_index)
+                    for (i_index, j_index) in comp_map if i_index < j_index]
```

Comment:
- you don't need brackets when you unpack variable: `for i_index, j_index in comp_map`
- you are mutating the value of `comp_map` give it a more descriptive name. I think the frist one can be called `full_comp_map` and the second one is `unique_comp_map`
- This is a place of possible confusion. You need more documentaion